include/crab/domains/symbolic_variable_eq_domain.hpp

```diff
@@ -0,0 +1,1031 @@
+#pragma once
+
+#include <crab/domains/abstract_domain.hpp>
+#include <crab/domains/abstract_domain_specialized_traits.hpp>
+#include <crab/domains/backward_assign_operations.hpp>
+#include <crab/domains/constant.hpp>
+#include <crab/domains/symbolic_variable_eq_domain.hpp>
+#include <crab/domains/term/term_operators.hpp>
+#include <crab/domains/union_find_domain.hpp>
+#include <crab/support/stats.hpp>
+
+#include <boost/optional.hpp>
+
+namespace crab {
+namespace domains {
+
+namespace symbolic_variable_equiality_domain_impl {
+class symbolic_var {
+  using this_domain_t = symbolic_var;
+
+public:
+  using var_id = uint32_t;
+
+  var_id var;
+
+  symbolic_var(var_id _var) : var(_var) {}
+  symbolic_var(const this_domain_t &o) = default;
+  symbolic_var(this_domain_t &&o) = default;
+  this_domain_t &operator=(const this_domain_t &o) = default;
+  this_domain_t &operator=(this_domain_t &&o) = default;
+
+  void write(crab_os &o) const { o << "#var" << var; }
+
+  bool operator==(this_domain_t o) const { return var == o.var; }
+
+  operator var_id() const { return var; }
+
+  friend class crab::crab_os &operator<<(crab::crab_os &o,
+                                         const this_domain_t &dom) {
+    dom.write(o);
+    return o;
+  }
+};
+
+struct symb_var_hash {
+  std::size_t operator()(const symbolic_var &k) const {
+    return std::hash<uint32_t>()(k.var);
+  }
+};
+
+template <class InputIt1, class InputIt2>
+bool set_is_absolute_complement(InputIt1 first1, InputIt1 last1,
+                                InputIt2 first2, InputIt2 last2) {
+  while (first1 != last1 && first2 != last2) {
+    if (*first1 < *first2)
+      ++first1;
+    else {
+      if (!(*first2 < *first1))
+        return false; // *first1 and *first2 are equivalent.
+      ++first2;
+    }
+  }
+  return true;
+}
+
+template <class SYMBVAR> SYMBVAR make_fresh_var_symbol() {
+  return SYMBVAR(term::term_op_val_generator_t::get_next_val());
+}
+
+template <class Domain> class equivalence_class {
```

Comment:
If `equivalence_class` is not going to be used outside of this file, you can hide it and make it a struct. Then you don't need to make getters and setters